codegen/src/api/storage.rs

```diff
@@ -71,25 +58,28 @@ pub fn generate_storage(
     })
 }
 
+/// Returns storage entry functions and alias modules.
 fn generate_storage_entry_fns(
     type_gen: &TypeGenerator,
     pallet: &PalletMetadata,
     storage_entry: &StorageEntryMetadata,
     crate_path: &syn::Path,
-    should_gen_docs: bool,
-    types_mod_ident: &syn::Ident,
 ) -> Result<(TokenStream2, TokenStream2), CodegenError> {
     let snake_case_name = storage_entry.name().to_snake_case();
     let storage_entry_ty = storage_entry.entry_type().value_ty();
-    let storage_entry_value_ty = type_gen.resolve_type_path(storage_entry_ty);
+    let storage_entry_value_ty = type_gen
+        .resolve_type_path(storage_entry_ty)
+        .expect("type should be present");
 
     let alias_name = format_ident!("{}", storage_entry.name().to_upper_camel_case());
     let alias_module_name = format_ident!("{snake_case_name}");
     let alias_storage_path = quote!( types::#alias_module_name::#alias_name );
 
     let storage_entry_map = |idx, id| {
         let ident: Ident = format_ident!("_{}", idx);
-        let ty_path = type_gen.resolve_type_path(id);
+        let ty_path = type_gen
+            .resolve_type_path(id)
+            .expect("type should be present");
```

Comment:
Infallible? Is it possible to for some type to not be in the PortableRegistry? I'm just asking wheter subxt should provide a better compile error than the expect.

Reply:
Before I think subxt generated a Codegen Error

Reply:
Should also be infallible, if we assume valid metadata.